nexus/reconfigurator/planning/src/system.rs

```diff
@@ -304,6 +306,31 @@ impl SystemDescription {
         Ok(self)
     }
 
+    /// Set Omicron zones for a sled.
+    ///
+    /// Returns an error if the sled is not found.
+    ///
+    /// # Notes
+    ///
+    /// While bootstrapping an example system, the order is:
+    ///
+    /// 1. Create a collection with sleds with no zones.
+    /// 2. From the collection, generate a blueprint with zones.
+    /// 3. Seed the system with zones from the blueprint.
+    ///
+    /// This method is used in step 3.
```

Comment:
Sorry about not being clear here! Yes, it means "when `Self::to_collection_builder` is called next, it will contain these Omicron zones for the sled".

I agree that describing the bootstrap process isn't very useful (you can discover this by looking for the callers of `sled_set_omicron_zones`), so I've dropped it.